src/agent/tests/register_test.cpp

```diff
@@ -1,12 +1,57 @@
 #include <register.hpp>
 
+#include <agent_info.hpp>
+
+#include <gmock/gmock.h>
 #include <gtest/gtest.h>
 
-TEST(RegistrationTest, RegistrationTest)
+using ::testing::_;
+using ::testing::MockFunction;
+using ::testing::Return;
+
+class RegisterTest : public ::testing::Test
+{
+protected:
+    const AgentInfo* agent;
+
+    MockFunction<std::optional<std::string>(
+        const std::string&, const std::string&, const std::string&, const std::string&)>
+        MockAuthenticateFunction;
+    MockFunction<http::status(const std::string& host,
+                              const std::string& port,
+                              const std::string& token,
+                              const std::string& uuid,
+                              const std::string& key,
+                              const std::optional<std::string>& name)>
+        MockRegistrationFunction;
+
+    ~RegisterTest()
```

Comment:
No need for a custom destructor if using smart ptr.

Reply:
Done!